vendor/bat-native-ledger/src/ledger_impl.cc

```diff
@@ -83,15 +83,12 @@ void LedgerImpl::Initialize() {
 }
 
 bool LedgerImpl::CreateWallet() {
-  if (initializing_)
+  if (initializing_) {
     return false;
+  }
 
   initializing_ = true;
-  if (initialized_) {
-    OnWalletInitialized(ledger::Result::LEDGER_ERROR);
-    return false;
-  }
-  bat_client_->registerPersona();
+  bat_client_->ShouldCreateWallet();
```

Comment:
`ShouldCreateWallet` is confusing as the naming suggests the function should return a bool, but instead the function also creates the wallet

Reply:
do you have suggestion for a name? Maybe just `CreateWallet`?

Reply:
`CreateWalletIfNecessary`

Reply:
fixed